$OpenBSD: patch-configure_in,v 1.3 2009/04/15 16:12:52 kili Exp $

XXX disabling PIE on arm is a workaround to let dbus work.

--- configure.in.orig	Tue Jan  6 19:30:26 2009
+++ configure.in	Tue Apr 14 21:58:52 2009
@@ -253,17 +253,20 @@ if test "x$GCC" = "xyes"; then
      ;;
   esac
 
-  case " $CFLAGS " in
-  *[\ \	]-fPIE[\ \	]*) ;;
-  *) if cc_supports_flag -fPIE; then
-        PIE_CFLAGS="-fPIE"
-        if ld_supports_flag -z,relro; then
-           PIE_LDFLAGS="-pie -Wl,-z,relro"
-        else
-           PIE_LDFLAGS="-pie"
+  case $host in
+  arm-*-openbsd*) ;;
+  *) case " $CFLAGS " in
+     *[\ \	]-fPIE[\ \	]*) ;;
+     *) if cc_supports_flag -fPIE; then
+           PIE_CFLAGS="-fPIE"
+           if ld_supports_flag -z,relro; then
+              PIE_LDFLAGS="-pie -Wl,-z,relro"
+           else
+              PIE_LDFLAGS="-pie"
+           fi
         fi
-     fi
-     ;;
+        ;;
+     esac
   esac
   
   ### Disabled warnings, and compiler flag overrides
@@ -282,10 +285,6 @@ if test "x$GCC" = "xyes"; then
   *[\ \	]-Wno-sign-compare[\ \	]*) ;;
   *) CFLAGS="$CFLAGS -Wno-sign-compare" ;;
   esac
-  case " $CFLAGS " in
-  *[\ \	]-Wno-pointer-sign[\ \	]*) ;;
-  *) CFLAGS="$CFLAGS -Wno-pointer-sign" ;;
-  esac  
   
   # http://bugs.freedesktop.org/show_bug.cgi?id=19195
   case " $CFLAGS " in
